pkg/api/validation/validation.go

```diff
@@ -2844,10 +2799,16 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	// Clear status
 	oldNode.Status = node.Status
 
+	// update taints
```

Comment:
Plus IIUC we're planning to allow changing Pod resource requests at some point which will get rid of this assumption. I guess it's better to do it sooner rather than later.